### ISSUE TITLE
System 7 display

### ISSUE THREAD

**Issue by patzik** [AmokSolderer/APC#24]
Not so much an issue, more a request.

I found your design of the replacement for the score display. I have a system 9 machine (comet) with 2 broken displays and would love to build a replacement for this. Would this display also work on this machine? Now for the request, my machine had the driver board with the edge connectors, could you PLEASE make a version of the driver board with this connection?
If there is anything I can do, just let me know, I don't know what design software you use, but I am familiar with eagle so I might be able to change the design if I had the source files.

**Comment by patzik** [comment]
Also, what kind of led segment displays did you use and where did you get them?

**Comment by AmokSolderer** [comment]
The Sys7Alpha displays on this page are only for use with the Arduino Pinball Controller. They would work in a Comet, but only with the APC installed.
The original System9 hardware cannot use these displays as they're alphanumeric and they use the System11 connectors (that's why there are no edge connectors).

If you just want to replace the displays and not the controller you would need something like this [http://baumfalk.net/Displays/sys7_en.html](url).
This is a design I did some 15 years ago, not sure whether I'm able to find the old design files.

**Comment by patzik** [comment]
Yeah, I noticed the alpa displays and the 'extra' connector today, but the link you posted looks promising! I can't etch pcb's so I'm unable to use those files, but looks like that's exactly what I need. I would really appreciate it if you can look for those files, it would be great if I can make some gerbers and have those boards made. I might look into the APC in the future, but I first want to get the displays working so my sons can play with their friends...

**Comment by patzik** [comment]
I tried all day to import the pdf's into eagle but the result is not something I can work with. The best thing would be the source files so I can also adjust the display boards if needed for other 7 segment display elements, maybe the cheaper Chinese ones.

I have my fingers crossed ;-)

**Comment by AmokSolderer** [comment]
I found some old design files, but they don't work with my actual KiCad version any more. Looks like I have to install an older version to open those files.
This is going to take a while I'm afraid.

**Comment by patzik** [comment]
Great that you found some of the files!
Could you send me the design files or put them on the GitHub, I can install an older version of kicad or convert the files and see what I can do with them.

**Comment by AmokSolderer** [comment]
The files are a mess. They have been migrated some time ago from a 15 year old Eagle version to KiCad 3. With a bit of rework I was now able to import them in KiCad 4. The Layout looks OK, just the board text is still missing. The schematic is still messy, but hopefully good enough to move forward to KiCad 5 for the final overhaul.
You can take a look in the V00.13 branch on GitHub. The Archive is in APC/DOC/Hardware/Sys7_Display, but I strongly recommend to wait until I had time to fix it.

**Comment by patzik** [comment]
I took a quick look, and I see what you mean with the schematic. I'd like to help, unfortunately I use eagle, so I'm not that familiar with kicad (although I have it installed). The board looks fine, so I guess it would be possible to generate gerbers from this or would that run into problems?
But the kingbright led's are hard to get here and ordering them from the us would be costly.
I found some similar led's in China (less than 0.40 euro each), and they should work just the same, but they have a different pinout, maybe you could make 2 display boards for the different led's? Here's a picture of their details.


![](https://ae01.alicdn.com/kf/HTB1jiSebAzoK1RjSZFlq6yi4VXak.jpg)


Thank you for looking into this, really appreciate it!

**Comment by patzik** [comment]
I was able to open the files with 4.0.7 and looks much better now, version 5 of kicad gives me noting but trouble :-(
Still having problems with switching from eagle to kicad, it's just so different, I don't understand how kicad works, so my hopes are one you...

**Comment by AmokSolderer** [comment]
I'm sorry, but I just don't have enough time to support different display types. Apparently it's going to be more than enough trouble to bring my old designs to KiCad5.

It would be a different story if you'd be able to find a display having a package width of less than 20mm. That's because the current displays are a bit too wide and need to be grinded down to the correct width which is kinda cumbersome.

**Comment by patzik** [comment]
I totally understand, and appreciate the time you put into this. I have looked for displays that are less than 20mm but can only find the 20mm ones. Do the 'commercial' reproduction displays also file down the segments or are they available in a narrower size?

**Comment by AmokSolderer** [comment]
They are available in China, but you have to order at least 1000 pcs.

**Comment by patzik** [comment]
Yes I see that the segments can be 19.05mm wide. I Can't find any dip packages in that size, what do you have any info on the ones that are available? I did however find smd packages that should work perfectly, they are also thinner so that should also help with 'distance between the display and the backglass. Did you look into these?
<img width="606" alt="Screenshot 2019-12-10 at 10 19 57" src="https://user-images.githubusercontent.com/32332590/70516299-0c325d80-1b37-11ea-871d-a173bded711a.png">

**Comment by AmokSolderer** [comment]
> Did you look into these?
No, I didn't actively look for a replacement for a long time. The Kingbright displays never failed me and grinding them down a bit is a bit cumbersome, but not that much work especially if you only have to do it every now and then.
The only thing that's bugging me is that I also did a design for a System11c display (2x16 alphanumeric), but never realized it because it requires displays with even less width. Therefore I'd be interested what kind of display you found and whether it is available as an alphanumeric type also.

I do now that a display called 8018SBO is being used for some pinball displays. Search Google for it and you will find some chinese pages offering them in quantities of 1000+. There's also a numerical version called 8015SBB.

**Comment by patzik** [comment]
They are ACSC08-51SEKWA (same brand), I found these here.
https://nl.mouser.com/Search/Refine?Keyword=ACSC08
But they are also available in China
https://nl.aliexpress.com/item/32715904139.html?spm=a2g0s.8937460.0.0.399e2e0eRfcL0o

As far as I can see they are not available in a 16 segment version.

Looking at the footprint of the ones I mentioned earlier (the dip version) and these, i guess it would be possible to design a board that can accommodate both if you use longer pads with the through hole. (just and idea)
I find kicad hard to use, do you know if it's possible to convert from kicad to eagle?

**Comment by AmokSolderer** [comment]
Are you located in the Netherlands? Then you might want to try Reichelt.de, as they have quite good prices. [https://www.reichelt.de/7-segment-anzeige-rot-20-3mm-gem-anode-sa-08-11-rt-p31575.html?&trstct=pol_0](url)

I also found the old Eagle files and put them to GitHub.

**Comment by patzik** [comment]
I am indeed from NL, and those are indeed reasonable prices so that might be an option.
How did you 'file down' the digits to make them narrower?

**Comment by patzik** [comment]
I have downloaded the eagle files, and after some work got them into the current version of eagle. Easier now to work on for me. I have 'played' a bit with the credit display, and have made one with the footprint of the Chinese 7-segment led's which will be easy to adapt to the smd ones and maybe one compatible with both. I am not the best in routing and just used the autorouter for now.
I don't know if you still use eagle, but could you take a look at it before I go any further?
If so, how can I send the files?

**Comment by patzik** [comment]
Looked at making a design compatible with both led variant. I thought they had the same pinout, but it seems I was wrong it's different so that's not an option.

**Comment by AmokSolderer** [comment]
> How did you 'file down' the digits to make them narrower?
I use a bench grinder, but a belt grinder would also do the lob.

> I don't know if you still use eagle, but could you take a look at it before I go any further?
I don't have a working Eagle installation any more.

**Comment by patzik** [comment]
The digits kinda look like a white plastic housing filled with some sort of epoxy, do you need to remove more than just the white part?
I adapted the board again and made a version for the kingbright smd led's. I think I will order some boards and just try and see what works best after the holidays.
I have a few more questions regarding the driver board, you uploaded various versions.
10, 10p, 11p, 20p can you remember what the differences are?
I see some difference between the 20p version in eagle and the 20p version in kicad, did you make a lot of changes after importing it into kicad.
The kicad version looks best, if I generate gerbers from this and maybe just change the display score and ball/credit design in eagle and generate those gerbers from there, maybe that would work? I see some resistors with a value of 20, is that 20kohms or 20 ohms?

**Comment by patzik** [comment]
Can you also give me the values of the capacitors?

**Comment by AmokSolderer** [comment]
> The digits kinda look like a white plastic housing filled with some sort of epoxy, do you need to remove more than just the white part?

You have to remove a little bit more. I always place the leftmost display (the millions digit) on a display board. This part doesn't have to be grinded.
Then I start grinding the next display component on it's left side (which will point to the already populated one). After the white plastic housing is gone, I try to put it next to the previous one. I stop grinding when the display components don't touch each other any more.

> 20p can you remember what the differences are?

Don't remember any more, but 20p is the one to use.

> I see some difference between the 20p version in eagle and the 20p version in kicad, did you make a lot of changes after importing it into kicad.

KiCad had some problems with the filled zones, so I had to change them. In the meanwhile I have completely imported the design to KiCad5 which required a lot of rework especially in the schematic. You can find the schematic and Gerber data on GitHub.

> I see some resistors with a value of 20, is that 20kohms or 20 ohms?

20 is 20 ohms.

> Can you also give me the values of the capacitors?

They are in the new schematic.

**Comment by patzik** [comment]
Looks great, and I can't thank you enough for doing this!

I see one tiny issue with the gerbers, there is silkscreen on the edge connectors.

The schematic and bom will help me with the parts, and I will definitely build this to fix my comet pinball machine. Just have to make up my mind of what displays/leds to use, the one you made and grind the segments or the one I made with the smd leds.

**Comment by AmokSolderer** [comment]
> Looks great, and I can't thank you enough for doing this!

You're welcome. I had to do it anyway sooner or later.

> I see one tiny issue with the gerbers, there is silkscreen on the edge connectors.

Fixed it and also removed most of the component values. Should be more readable now.

**Comment by patzik** [comment]
I’m on a ski trip right now, will check it out when I get back.

**Comment by patzik** [comment]
I took a quick look at the gerbers on my ipad, and was just wondering what side the parts go? Most silkscreening  is on the topside, but parts like the headers for the ribbon cable I assume go on the backside. So what about the other parts?

**Comment by AmokSolderer** [comment]
Most of the components are through-hole, which means they are placed on the top but soldered on the bottom layer.
The only components mounted on the bottom layer are four SO-8 ICs.

**Comment by patzik** [comment]
Also the pin headers? Because when I look at the orientation of the board and place it in the pin, the components and headers would be on the side that goes against the door, so not like the original board away from the door.

**Comment by AmokSolderer** [comment]
Also the pin headers. The placement of the original board may be different.

I have uploaded a shot of my board to GitHub. There you can see that almost everything is placed on the top side.

**Comment by patzik** [comment]
I was about to order the boards, but noticed you only have gerbers for the driver board and the alphanumeric display, not the numeric display. I still have a copy of the eagle files you had on the GitHub (they are gone now). Can I generate the gerbers from these files, will they work with the gerbers of the driver board on GitHub? And one more question regarding the led segments, just to be sure, are they common cathode or common anode?
Oh and the picture makes it all very clear, thanks for posting that! I think the connectors won't be in the way, but I might solder the big cap on the other side, it looks like it will be in the way of the backglass.

**Comment by AmokSolderer** [comment]
I have uploaded the numeric displays to GitHub. I recommend to use them instead of the old eagle files because I changed the footprint of the pinheader to one which is easier to solder.

The LED Displays have to be common anode.

**Comment by patzik** [comment]
Ok thanks, I will order the boards and will keep you updated. Chinese new year now so will take a bit longer...

**Comment by patzik** [comment]
Regarding the alphanumeric displays, I might have found something.
Minimum order 100, and at a very good price!
https://www.alibaba.com/product-detail/blue-0-8-inch-led-16_60310092939.html?spm=a2700.galleryofferlist.0.0.5f0f19740KWb3h

**Comment by patzik** [comment]
With all that is happening, I have a quite a bit more time at home so I ordered the boards a while ago and they finally arrived. Still waiting for some parts, but received  the 7-segment led's today.
I had already printed a small saw table for my dremel to cut them to size, so was eager to get started. I didn't order the led's from reichelt, but saw a good deal of 100 pieces for about 30 euro's and they were the same (although from a different brand). Unfortunately they send me the wrong ones :-( They are common anode but have the pinout from the ones I mentioned earlier.
![IMG_2020](https://user-images.githubusercontent.com/32332590/78826435-c51f8080-79e1-11ea-9ba4-2645a5de7f58.jpeg)

**Comment by patzik** [comment]
The seller is investigating it right now, and I might get the correct ones, but looks like I will have to try to redesign the score displays after all...

**Comment by patzik** [comment]
Well, I finally got the right 7segment led displays in, and all the required chips. Was very happy, because now I could finally start soldering everything together. I finished everything today and was ready to test. Disappointment... it doesn't work I will post pictures of what I get, but basically it's digit 1off 2on 3off 4on 5on 6on 7on and one of the dots is flashing like it should on the display.
I replaced all chips with different ones, but the result is the same. Really hope you can help me out.

**Comment by patzik** [comment]
![IMG_2078](https://user-images.githubusercontent.com/32332590/81432556-6c1f4580-9163-11ea-9f2c-60cf8d38b896.jpeg)
![IMG_2079](https://user-images.githubusercontent.com/32332590/81432560-6f1a3600-9163-11ea-9d57-b8a184012892.jpeg)
![IMG_2080](https://user-images.githubusercontent.com/32332590/81432565-717c9000-9163-11ea-9f94-632122ea773c.jpeg)

**Comment by AmokSolderer** [comment]
Missing digits are usually caused by bad solder joints at the strobe driver transistors. These are the SMD ICs where bad soldering happens very easily.
To find the right transistor you have to identify which strobe doesn't work and then look for the corresponding transistor.

Let's use the credit display as an example.
Looking at the schematics you can see that the credit display is using STR1 and STR9. As the first digit fails you problem has to be STR1. According to the schematics STR1 is handled by Q5A, so check this transistor for bad soldering. You should download the IRF7316 datasheet to identify the correct pins, since there're two transistors in each package.

**Comment by patzik** [comment]
I think I have found the problem with some of the strobe signals not working and the digits that are out because of this. What I still can't figure out is why all segments are continuously lit.
Can you help me out with what to look for? What I tried so far, I put the machine in test mode and into the display test with one number displaying. I thought with this, the outputs of the segment decoder will always be the same as the segments don't change and are the same for each digit.
When I measure the output of the decoder all outputs measure about 3.7v. Shouldn't the outputs for the segments that shouldn't be lit be 0v? If so what could cause this? I tested several decoder chips and the result is always the same.

**Comment by patzik** [comment]
Hi I just read your reply, somehow I didn’t see it before posting the above. By coincidence this was the exact procedure I did last night. And after a replacement and some more soldering all digits of the credit display are now working. Don’t know if it was a faulty irf7316 or soldering but will do some more work to get the score displays all working and see what is the cause.
Problem with the decoder remains, hope you can help me with faultfinding this.

**Comment by patzik** [comment]
One more thing, as the komma’s work as intended (they flash in attract mode) I think it is in the bcd part of the circuit that the fault is.

**Comment by AmokSolderer** [comment]
No, it doesn't matter if all displays are connected or not.
Is it possible that you have the wrong BCD Decoder ICs?
The datasheet of the correct IC should be this one:
https://cdn-reichelt.de/documents/datenblatt/A240/MOS4511_SMD4511_SMD4511-S%23NSC.pdf

You should check whether the Input Signal are as shown in the truth table. For example the wrong Level at pin 3 (LT) would turn on all LED segments.

**Comment by patzik** [comment]
I decided to start from scratch and build another driver board, as I have 2 extra pcb's.
Unfortunately the same problem. I also resoldered the 7316's on the 1st board as still not all digit's light up. Then I noticed a 'similarity' between the boards, if I look at player 3 and 4 for instance, these use the same BCD decoder. The digits that light up (as an 8)on one display are blank on the other display. The same is for the 2nd board. This can't be a coincidence I think. However on the digits that are blank, if a komma is used, this also doesn't flash. On the ones that light up, when a komma is used, it flashes as it should.
I'm trying to find the logic sense behind all, but just can't figure it out.
I use CD4511BE ic's.

**Comment by patzik** [comment]
I checked LT BI and LE and these signals are as they should. I checked the outputs with a multimeter (can't find my probe right now) and they seem to switch between 2 voltages about 3.1 and 3.7 so never 0 or close to it. The chip seems to be doing something as it switches between 2 'levels' but these both look like values that would light the segments (as they do). What should be a normal output or input into the BCD as when I look into the inputs in testmode I would expect 0 or 5v which I also don't get.

**Comment by patzik** [comment]
I did some more measuring, and the input levels are actually as what I would expect for logic levels. It's the outputs that still remain at levels that I cannot 'explain'.

**Comment by patzik** [comment]
Turns out, that some of the digits not working in relation to the players was a coincidence. I did some more measuring etc today and started to replace the 7316's related to the digits not working. Don't know if they ones I replaced are really defective but in the end I now have a board where all the digits light up.
Leaves me still with the problem that all segments are continuously lit.
So what I tried is removing both 4511's and connected a segment output to 5v and this lights up the related segment.
Also tried is removing the 2803 and connected a segment to ground and this lights up the related segment.
So problems seems to be with the 4511's, their output is always around 4v so never goed low. I tried 10 ic's they can't all be bad can they?

**Comment by AmokSolderer** [comment]
No, they can't all be bad. If you're sure that you have the right ICs then may be your pinball MPU is not doing what you're expecting?
I think to check that you should unplug the BCD-IN1 connector and apply some BCD code manually. This way you can check your 4511 devices without any interference from your MPU.

BTW, I doubt that any of your 7316 is bad also. No device leaves the fab untested and you're not the first one having problems to solder SO-8 packages.

**Comment by patzik** [comment]
The original display is working fine (although 2 displays are outgassed), so I assume the MPU is also working fine. I connected a normal led with a resistor to the outputs of the 4511's and I can see the led although continuously lit changing brightness just a bit when the segment is turned on/off so it looks to be working but the low state is just not 0v. I will try to find new 4511's tomorrow, and see if that makes a change.
As far as the 7316's, I don't think it's my soldering unfortunately. I have soldered many projects with tqfp packages and 0603 parts by hand, so compared to these the 7316 is huge ;-)

**Comment by patzik** [comment]
Well, made some progress today. Luckily there is an electronics store really close to my home with a huge assortment. They had 4511's from different brands and I bought a set of motorola's and a set of philips. The motorola's didn't light all segments, it was more the opposite, maybe one segment lit up and didn't go out so no success. The philips however worked! All segments work perfectly now. I can't really explain this, and am really curious why some brands work and some brands don't, but I'm happy I found a working set.
Only one thing left to fix now, because I still have a problem with some digit's not lighting up. Seems the irf7316's are also hit and miss. I have gotten to the point that all digits except digit one on display 1 and 3 are working. I have changed the related 7316 (Q1) about 5 times now and can't seem to get it to work. Strangely enough there is a voltage on the common pin of the digit, so you would think it would work. I tried different displays but this made no difference. The outputs are 5+6 and 7+8, somehow if I short the 2 pairs the 1'st digit does light up but I don't think it would display correctly (only tried test mode where all numbers are the same).
I also swapped around the 4050's but this also made no difference and the contact on the edge connector is good and the digit does work with the original displays so seems like another strange problem...
Will try some more options, and hope to get it to work.
I designed a 'cover' for the displays where only the segments and the 2 komma's are visible, works out really nice. Have to finetune it just a bit, are you interested in adding these to the GitHub when finished?

**Comment by patzik** [comment]
Success!!

I still don't know why and how, but swapping some more with the irf7316 finally ended up in a working combination. I'm 100% sure it's not the soldering and will do some more 'troubleshooting' with the 2nd board, as it is build already, and I'm very curious about the cause. But not touching this board again, this one stays in!
Thank you so much for your help, hope I can return the favour someday. Oh and If you like to add the stl for the 'cover' just let me know.
![IMG_2088](https://user-images.githubusercontent.com/32332590/81732466-d2bca000-9490-11ea-9cc8-4a23d2bb05eb.jpeg)

**Comment by AmokSolderer** [comment]
That's very strange. I've used dozens of IRF devices over the years (the APC uses them also) and never had a bad one.
Probably your cover does only fit on your special display elemnts or do you think it would fit on the Reichelt ones also?

**Comment by patzik** [comment]
The size of the Reichelt ones are the same so it should also fit the same. I am not sure if the cutouts for the segments and komma are at the exact same place though, and I don't have one of those elements to testfit. If I can export a step file it will be very easy to change though as it is a very simple model. You could just shift the four corners of the cutout to where they should be and also the komma.
I have done some more soldering and desoldering of the irf's and sadly I think I got a batch of crapy ones from china, over half of they are partial or completely not working, have to say out of all the parts I ordered from there, this is the worst case of non working parts. Unfortunately the local electronics shop here doesn't have smd components, so I'll add them to my list for next order at mouser or reichelt.
